pointcloud/kdtree.go

```diff
@@ -135,6 +136,45 @@ func ToKDTree(pc PointCloud) *KDTree {
 	return t
 }
 
+// ToBalancedKDTree
+func ToBalancedKDTree(pc PointCloud) *KDTree {
```

Comment:
Attempted to make a function to accept a pointCloud and create a balanced KDTree but the constructor on 172 likely takes too long since we're already iterating through all the points in the iterate function. The call on 153 also doesn't help because contains is a linear function. I think slices are also just not stored continuously in memory, so it's all kind of rough.